stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h

```diff
@@ -44,8 +44,9 @@
 #define HAL_DCMI_MODULE_ENABLED
 #define HAL_DMA_MODULE_ENABLED
 #define HAL_DMA2D_MODULE_ENABLED
+/* ETH driver is chosen by CMake/Kconfig */
 /* #define HAL_ETH_MODULE_ENABLED */
-#define HAL_ETH_LEGACY_MODULE_ENABLED
+/* #define HAL_ETH_LEGACY_MODULE_ENABLED */
```

Comment:
See above:
```suggestion
if(CONFIG_ETH_STM32_HAL_API_V2)
  #define HAL_ETH_MODULE_ENABLED
else
  #define HAL_ETH_LEGACY_MODULE_ENABLED
endif()
```

Reply:
CMake syntax in a header file? But I get it...